### PR TITLE
Better error message when OCaml compiler not found for native compute

### LINKED ISSUE #6699: native_compute error message when not finding the OCaml compiler should be improved (and not be an anomaly).
#### Version

> The Coq Proof Assistant, version 8.7.1 (December 2017)
> compiled on Dec 18 2017 21:18:25 with OCaml 4.04.2

#### Operating system

> `Linux 4.9.76 #1-NixOS SMP Wed Jan 10 08:29:55 UTC 2018 x86_64 GNU/Linux`

#### Description of the problem

Below code fails to evaluate in `native_compute` but works fine in `vm_compute`

Produces error:
> Error: Anomaly "Compilation failure."
> Please report at http://coq.inria.fr/bugs/. 

```coq
Section Demo.
  Class P m := {propv : nat -> m}.
  Inductive T := t_p : nat -> T.
  Instance tp : P T := {|propv := t_p|}.

  Parameter p : nat.
  
  Eval vm_compute in (propv 5).
  Eval vm_compute in (propv p).
  Eval native_compute in (propv p). (* compiler error *)
  Eval native_compute in (propv 5). (* compiler error *)
End Demo.
```

### PR DESCRIPTION
**Pull request by maximedenes**
Fixes #6699

### CODE REVIEW

**Comment by maximedenes** [comment]
This is a trivial improvement. Any takers? @silene @ejgallego ?

**Comment by silene** [comment]
So, just to be clear, you turned a warning and an anomaly into an error, right? Seems fine with me. I am not accepting the pull request yet, since only the last commit is relevant.

**Comment by maximedenes** [comment]
> I am not accepting the pull request yet, since only the last commit is relevant.

Oh! This is a mistake, you're right, will fix.

**Comment by maximedenes** [comment]
Should be good now.

File: kernel/nativelib.mli [hunk 1]
```diff
@@ -21,9 +21,9 @@ val load_obj : (string -> unit) ref
 
 val get_ml_filename : unit -> string * string
 
-val compile : string -> global list -> profile:bool -> bool * string
+val compile : string -> global list -> profile:bool -> string
 
-val compile_library : Names.DirPath.t -> global list -> string -> bool
+val compile_library : Names.DirPath.t -> global list -> string -> unit
```
Comment (ejgallego) [inline kernel/nativelib.mli]: Please add a brief `docstring` while you are at it.

**Review by ejgallego — APPROVED** [review]
Looks good.

**Comment by ejgallego** [comment]
@maximedenes I'll merge this today as there is no reason to delay; however it'd be great if you could take 10 minutes to write the docstring for the function; I am slowly pushing for more API documentation and IMO in this case it would help to have a docstring.

**Comment by ejgallego** [comment]
> however it'd be great if you could take 10 minutes to write the docstring for the function;

Ping @maximedenes , if you don't have time I can just merge as-is.

**Comment by maximedenes** [comment]
> Ping @maximedenes , if you don't have time I can just merge as-is.

I'm on holiday, so feel free to proceed.

**Comment by ejgallego** [comment]
> I'm on holiday, so feel free to proceed.

Ok, will merge then [or you hint me briefly on what `compile` and `compile_library` arguments are I'll add the docstring myself]

**Pull request merged**